### PR TITLE
USHIFT-850: Document rebase job internals and testing

### PR DESCRIPTION
**Pull request by pmtk**

### CODE REVIEW

File: docs/rebase.md [hunk 1]
```diff
@@ -157,3 +157,148 @@ git commit -m "update buildfiles"
...
+
+To reduce a need of Pull Request synchronization between repositories, keep logic together, and allow for easier testing and developing, all scripts related to rebasing are residing in `scripts/auto-rebase/` directory in this repository.
+
+[Job's logic is to only execute](https://github.com/openshift/release/blob/1976df80259f0357c0842700a70a64433988892a/ci-operator/step-registry/openshift/microshift/rebase/openshift-microshift-rebase-commands.sh)
+`scripts/auto-rebase/rebase_job_entrypoint.sh` which gathers necessary arguments and passes them to `rebase.py` and is implemented in a way expecting to be executed in the job's container but can be a guidance on what arguments provide to `rebase.py` locally.
+
+
+#### Getting release image build references
+
+Rebase procedure expects references to AMD64 and ARM64 release images.
```
Comment (dhellmann) [inline docs/rebase.md:239]: It needs the CSI image, too, now that #1108 landed, right?

**Review by dhellmann — COMMENTED** [review]
Nice doc, thank you. I'd like to make sure another member of the team reviews the text in case they have questions, so I'll leave it open to give time for that.

File: docs/rebase.md (hunk 1 above)
Comment (pmtk) [reply docs/rebase.md:239]: That is correct

**Comment by fzdarsky** [comment]
Very clear IMO. Thank you for documenting away that tribal knowledge!

File: docs/rebase.md [hunk 2]
```diff
@@ -157,3 +157,148 @@ git commit -m "update buildfiles"
...
+They can be obtained from Release Status pages: (AMD64)[https://amd64.ocp.releases.ci.openshift.org/] and (ARM64)[https://arm64.ocp.releases.ci.openshift.org/] - navigate to section with nightly image builds for version that is currently worked on and pick latest approved for both architectures.
+These references are passed to `rebase.py` using `AMD64_RELEASE` and `ARM64_RELEASE` environment variables, for example:
+```
+AMD64_RELEASE=registry.ci.openshift.org/ocp/release:4.13.0-0.nightly-2023-01-27-165107 \
+ARM64_RELEASE=registry.ci.openshift.org/ocp-arm64/release-arm64:4.13.0-0.nightly-arm64-2023-01-30-010253 \
+./scripts/auto-rebase/rebase.py
```
Comment (ggiguash) [inline docs/rebase.md:245]: Need to fix the command to include LVMS

File: docs/rebase.md [hunk 3]
```diff
@@ -157,3 +157,148 @@ git commit -m "update buildfiles"
...
+
+```bash
+TOKEN=ghp_... \
+ORG=openshift \
+DRY_RUN=y \
+REPO=microshift \
+AMD64_RELEASE=registry.ci.openshift.org/ocp/release:4.13.0-0.nightly-2023-01-27-165107 \
+ARM64_RELEASE=registry.ci.openshift.org/ocp-arm64/release-arm64:4.13.0-0.nightly-arm64-2023-01-30-010253 \
+./scripts/auto-rebase/rebase.py
+```
```
Comment (ggiguash) [inline docs/rebase.md:265]: Need to add LVMS argument

File: docs/rebase.md [hunk 4]
```diff
@@ -157,3 +157,148 @@ git commit -m "update buildfiles"
...
+
+    git remote add pmtk https://github.com/pmtk/microshift.git
+    git fetch pmtk
+    git switch --track pmtk/csi-rebase-script
+
+    # export DRY_RUN=y
+    
+    ./scripts/auto-rebase/rebase_job_entrypoint.sh
+    
+    git diff csi-rebase-script
```
Comment (ggiguash) [inline docs/rebase.md:297]: Can we update the script to get the user name from the command line, or at least, to set it in a var in one place?

File: docs/rebase.md [hunk 5]
```diff
@@ -157,3 +157,148 @@ git commit -m "update buildfiles"
...
+Script also can be ran against a private fork in non-dry run which is helpful to verify that the communication with remote repository is as expected and so is created PR.
+In such case `ORG` env var needs to be set to GitHub username:
+
+```bash
+TOKEN=ghp_... \
+ORG=USER_NAME \
+REPO=microshift \
+AMD64_RELEASE=registry.ci.openshift.org/ocp/release:4.13.0-0.nightly-2023-01-27-165107 \
+ARM64_RELEASE=registry.ci.openshift.org/ocp-arm64/release-arm64:4.13.0-0.nightly-arm64-2023-01-30-010253 \
+./scripts/auto-rebase/rebase.py
```
Comment (ggiguash) [inline docs/rebase.md:276]: Need to add LVMS argument

**Review by ggiguash — CHANGES_REQUESTED** [review]

File: docs/rebase.md (hunk 1 above)
Comment (pmtk) [reply docs/rebase.md]: LVMS added

File: docs/rebase.md (hunk 2 above)
Comment (pmtk) [reply docs/rebase.md:266]: LVMS added

File: docs/rebase.md (hunk 3 above)
Comment (pmtk) [reply docs/rebase.md:287]: LVMS added

File: docs/rebase.md (hunk 5 above)
Comment (pmtk) [reply docs/rebase.md:299]: LVMS added

File: docs/rebase.md (hunk 4 above)
Comment (pmtk) [reply docs/rebase.md]: This snippet shows example modifications to https://github.com/openshift/release/blob/master/ci-operator/step-registry/openshift/microshift/rebase/openshift-microshift-rebase-commands.sh which isn't intended to be executed locally (even more so in interactive environment), but added possibility to override ORG and REPO (rebase_job_entrypoint.sh) and reworded this section

**Pull request merged**